app/models/user.rb

```diff
@@ -1,6 +1,17 @@
 class User < ApplicationRecord
   has_many :posts, dependent: :destroy
+  has_many :active_relationships, class_name: 'Relationship', foreign_key: 'follower_id', dependent: :destroy
+  has_many :passive_relationships, class_name: 'Relationship', foreign_key: 'followed_id', dependent: :destroy
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
   validates :email, presence: true
+
+  def follow(user)
+    active_relationships.create(followed_id: user.id)
+  end
```

Comment:
idしか使っていないので、引数をidにしてもいいかなとおもった。呼び出し側でもuserオブジェクトを作らなくて済むし。
```suggestion
  def follow(user_id)
    active_relationships.create(followed_id: user_id)
  end
```